mne/export/_edf.py

```diff
@@ -97,20 +63,32 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
 
     ch_names = [ch for ch in raw.ch_names if ch not in drop_chs]
     ch_types = np.array(raw.get_channel_types(picks=ch_names))
-    n_channels = len(ch_names)
     n_times = raw.n_times
 
+    # get the entire dataset in uV
+    data = raw.get_data(units=units, picks=ch_names)
+
     # Sampling frequency in EDF only supports integers, so to allow for
     # float sampling rates from Raw, we adjust the output sampling rate
     # for all channels and the data record duration.
     sfreq = raw.info["sfreq"]
     if float(sfreq).is_integer():
         out_sfreq = int(sfreq)
         data_record_duration = None
+        # make non-integer second durations work
+        if pad_width := int(np.ceil(n_times / sfreq) * sfreq - n_times):
```

Comment:
Are we okay with using the walrus operator?

Reply:
Yes!

Reply:
But regarding this condition, would it maybe be better to make the comparison more explicit @hofaflo?

Reply:
I don' really have a preference here, so I'd like to let you decide (also for the [second walrus](https://github.com/hofaflo/mne-python/blob/edfio/mne/export/_edf.py#L199))!

Reply:
> But regarding this condition, would it maybe be better to make the comparison more explicit @hofaflo?

Yes that's what I meant, I don't think it helps readability in this specific case. But I'll let you decide

Reply:
I'd keep it, but I'd make the comparison more explicit by adding `> 0`.

Reply:
Same for the second one, what exactly are we comparing here? `if meas_date := raw.info["meas_date"] is not None:` maybe?

Reply:
> the [second walrus](https://github.com/hofaflo/mne-python/blob/edfio/mne/export/_edf.py#L199)

and leave this as-is?

Reply:
+1 for @cbrnr's suggestions